assets/javascripts/discourse/initializers/extend-for-assigns.js

```diff
@@ -478,7 +478,13 @@ function initialize(api) {
             assignedPath = `/t/${topic.id}`;
           }
           const icon = iconHTML(assignee.username ? "user-plus" : "group-plus");
-          const name = assignee.username || assignee.name;
+          let name;
+          if (siteSettings.prioritize_full_name_in_ux) {
+            name = assignee.name;
```

Comment:
you'll want this to fall back to `username` in case there's no full name entered